### ISSUE TITLE
Unable to install Resourcepacks or Shaderpacks when Minecraft is already running

### ISSUE THREAD

**Issue by DelofJ** [MultiMC/Launcher#4745]
### Operating System

Windows

### Description of bug

When you launch Minecraft, if you want to add Resourcepacks or Shaderpacks through the launcher, you need to close the game to do it, but unlike installing mods, you can add Shaderpacks and Resourcepacks while the game is running

### Steps to reproduce

1. Launch an instance in MultiMC
2. Go to the Resourcepacks/Shaderpacks tab
3. Drag a Resourcepacks/Shaderpacks into the tab
4. You can't do it

### Suspected cause

The launcher prevents us from doing this

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by peterix** [comment]
This is entirely intentional.

**Comment by DelofJ** [comment]
Ok but why it is intentional ?
ik we can't for mods because installing mods while Minecraft is running
But why we can't install resourcepacks/shaderpacks while Minecraft is running ?

**Comment by NovaLynxie** [comment]
> "Ok but why it is intentional? ik we can't for mods because installing mods while Minecraft is running But why we can't install resource packs/shader packs while Minecraft is running?"

Minecraft itself already supports drag & drop of resource packs while in the resource pack settings menu as well as open pack folder buttons to access said directories as shown here:
![image](https://user-images.githubusercontent.com/21307676/174321479-4466c33c-dd20-4c5c-94b8-9cabe14dcece.png)
Optifine and Iris shader menus both already have their own open pack folder button as well as dragging and dropping files.
Optifine Shaders:
![image](https://user-images.githubusercontent.com/21307676/174321514-f81a7feb-822a-4c59-aabc-3c75b7768129.png)
Iris Shaders (Fabric):
![image](https://user-images.githubusercontent.com/21307676/174321528-85c11d3c-9813-4828-ae9b-b82332ba28f7.png)

>  "but unlike installing mods, you can add Shaderpacks and Resourcepacks while the game is running"

You have basically defeated your own argument of why this should be a thing, you quite literally asked for a feature that Minecraft and shader mods themselves **already** provide while the game is running as shown above.
It is literally just a couple of clicks within the game's options menu to do exactly what you are asking.

**Comment by DelofJ** [comment]
> Minecraft itself already supports drag & drop
Ok, but pls just note that not all versions of Minecraft support drag & drop, that's why I'm asking why the launcher can't do it when Minecraft is already launched

**Comment by NovaLynxie** [comment]
Fair point but, the open resource/shader pack folder buttons exist in all versions that support resource/shader packs to allow changing these on the fly without having to close the game window anyway.
It's really not that hard to understand why this is intentional when Minecraft already has other features there to do exactly what you have asked already built-in.
As I see it, the launcher locks out any and all changes to prevent the user from changing things they probably shouldn't which could potentially break and /or corrupt the game instance while it is running.
You would not intentionally remove a resource/shader pack while it is loaded in-game while it's still running, would you?
And even if you did, most if not all operating systems would disallow you from editing/removing those files while Minecraft is still using them anyway.

**Comment by DelofJ** [comment]
In case you misunderstood, at no time I’m talking about removing a resourcepack while it is loaded, I'm talking about removing or adding one while it is not  loaded by the game. And I don't know if you are aware but adding a pack while the game is running will not corrupt it, you even said there are buttons to open the resource packs/shader packs folder when Minecraft is running
I know these buttons exists, so what’s the utility of these tabs in the launcher ?

**Comment by NovaLynxie** [comment]
> In case you misunderstood, at no time I’m talking about removing a resourcepack while it is loaded, I'm talking about removing or adding one while it is not loaded by the game. And I don't know if you are aware but adding a pack while the game is running will not corrupt it, you even said there are buttons to open the resource packs/shader packs folder when Minecraft is running I know these buttons exist, so for you what’s the utility of these tabs in the launcher ?

Look bud, the developer said it's intentionally meant to be this way meaning it's a feature and that is how the launcher works and will work.

For me, I only see it as a quick way of editing/adding/removing resource packs, shader packs, mods, etc. while the game isn't running to save time doing so before loading up the game. I'm sure no one wants to wait up to 10 minutes to load a modpack just to change a resource pack setting after it is loaded.
And yes I'm aware it won't outright corrupt anything, that wasn't my point however in my previous comment.
Personally, I don't see a need for this to be altered in this way when other options already exist as I have stated plenty of times as it just seems like more work for the developer's end for a feature that just seems redundant.
Simply put, it is designed to stop interacting with the client instance files period to avoid any unnecessary changes from its end while the game is running. It won't stop the user from messing with the files directly it just won't allow users to do any changes to them from its end.

I'm not going to keep going around in circles trying to explain this to you any further.

**Comment by peterix** [comment]
Think about it as guard rails on a bridge. You can still jump over them if you want to, but they are there for a reason - to keep the 99.99% who don't want to go over the guard rails from falling.

I am not changing this, it would go against the design principles of the launcher.
